pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -204,50 +318,102 @@ func setDefaultDeploymentSpec(spec *appsv1.DeploymentSpec) {
+			// Initialize Limits map if it's nil
+			if container.Resources.Limits == nil {
+				deployment.Spec.Template.Spec.Containers[i].Resources.Limits = make(map[corev1.ResourceName]resource.Quantity)
+			}
 
-		// To avoid the conflict between HPA and Deployment,
-		// we need to remove the Replicas field from the deployment spec
-		modDeployment := r.Deployment.DeepCopy()
-		modDeployment.Spec.Replicas = nil
+			// Assign the tensorParallelSize value to the GPU "nvidia.com/gpu" resource limits
+			deployment.Spec.Template.Spec.Containers[i].Resources.Limits[constants.NvidiaGPUResourceType] = resource.MustParse(tensorParallelSize)
 
-		modJson, err := json.Marshal(modDeployment)
-		if err != nil {
-			return nil, err
+			// Initialize Requests map if it's nil
+			if container.Resources.Requests == nil {
+				deployment.Spec.Template.Spec.Containers[i].Resources.Requests = make(map[corev1.ResourceName]resource.Quantity)
+			}
+
+			// Assign the tensorParallelSize value to the GPU "nvidia.com/gpu" resource requests
+			deployment.Spec.Template.Spec.Containers[i].Resources.Requests[constants.NvidiaGPUResourceType] = resource.MustParse(tensorParallelSize)
+			break
 		}
-		// Generate the strategic merge patch between the current and modified JSON
-		patchByte, err := strategicpatch.StrategicMergePatch(curJson, modJson, appsv1.Deployment{})
+	}
+}
+
+// Reconcile ...
+func (r *DeploymentReconciler) Reconcile() ([]*appsv1.Deployment, error) {
+	for _, deployment := range r.DeploymentList {
+		// Reconcile Deployment
+		checkResult, _, err := r.checkDeploymentExist(r.client, deployment)
 		if err != nil {
 			return nil, err
 		}
+		log.Info("deployment reconcile", "checkResult", checkResult, "err", err)
 
-		// Patch the deployment object with the strategic merge patch
-		opErr = r.client.Patch(context.TODO(), deployment, client.RawPatch(types.StrategicMergePatchType, patchByte))
+		var opErr error
+		switch checkResult {
+		case constants.CheckResultCreate:
+			opErr = r.client.Create(context.TODO(), deployment)
+		case constants.CheckResultUpdate:
+			curJson, err := json.Marshal(deployment)
+			if err != nil {
+				return nil, err
+			}
 
-	default:
-		return deployment, nil
-	}
+			// To avoid the conflict between HPA and Deployment,
+			// we need to remove the Replicas field from the deployment spec
+			modDeployment := deployment.DeepCopy()
+			modDeployment.Spec.Replicas = nil
 
-	if opErr != nil {
-		return nil, opErr
-	}
+			modJson, err := json.Marshal(modDeployment)
+			if err != nil {
+				return nil, err
+			}
+			// Generate the strategic merge patch between the current and modified JSON
+			patchByte, err := strategicpatch.StrategicMergePatch(curJson, modJson, appsv1.Deployment{})
+			if err != nil {
+				return nil, err
+			}
 
-	return r.Deployment, nil
+			// Patch the deployment object with the strategic merge patch
+			opErr = r.client.Patch(context.TODO(), deployment, client.RawPatch(types.StrategicMergePatchType, patchByte))
```

Comment:
shouldn't you also get the updated deployment as the `deploymentList` is returned?

Reply:
my point is, you patch the deployment and do not return the patched deployment as part of the returned DeploymentList...

Reply:
I'm not sure if I fully grasp your point. The original logic does not return the patched deployment because, as I mentioned earlier, it's a temporary measure to apply a patch and avoid conflicts with the Horizontal Pod Autoscaler (HPA). Do you think I should reconsider the original logic?